### ISSUE TITLE
Use litestream.io expanded config versus replica_url

### ISSUE THREAD

**Issue by richjdsmith** [fractaledmind/litestream-ruby#10]
Any interest in me putting together a pull request to refactor the gem to expand out the replica url to give the option to set `bucket`, `endpoint`, `region` as env variables?

 It would allow users to use Cloudfare R2 for replication (litestream requires region to work with R2)

**Comment by fractaledmind** [comment]
Yes, but branch off of #7 and not main. I will be finishing that branch soon, and it is starting to move in a similar direction. So we can avoid future conflicts if you start there
